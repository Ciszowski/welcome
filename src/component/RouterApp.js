import React from 'react';
import { BrowserRouter as Router, Routes, Route, Navigate } from 'react-router-dom';
import Welcome from './welcome/Welcome';

function RouterApp()
{
    return (
        <React.Fragment>
            <Router>
                <Routes>
                    <Route exact path="/home" element={<Welcome/>} />
                    <Route path="/" element={ <Navigate replace to="/home" />} />
                </Routes>
            </Router>
        </React.Fragment>
    )
};

export default RouterApp;