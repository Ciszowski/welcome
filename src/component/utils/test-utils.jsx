import React from 'react'
import { render } from '@testing-library/react'
import { configureStore, combineReducers } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'
import welcomeRX from '../../reducers/welcomeRX';
import { applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';

const reducer = combineReducers({ welcomeRX });
const devTools = process.env.NODE_ENV === 'development' ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__() : null

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({reducer}, compose(applyMiddleware(thunk), devTools)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}