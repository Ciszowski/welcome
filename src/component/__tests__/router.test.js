import React from 'react'
import RouterApp from '../RouterApp'
import { createMemoryHistory } from "history";
import { renderWithProviders } from '../utils/test-utils';


describe('ButtonLogin', () => {
    test('should pass', () => {
        const history = createMemoryHistory({ initialEntries: ['/home'] });
        const { asFragment } = renderWithProviders(
            <RouterApp history={history} />
        )
        expect(history.location.pathname).toBe('/home');
        expect(asFragment()).toMatchSnapshot()
    });
  });