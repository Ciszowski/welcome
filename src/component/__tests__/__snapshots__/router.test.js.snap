// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLogin should pass 1`] = `<DocumentFragment />`;
